test-lib-openshift.sh

```diff
@@ -368,17 +377,97 @@ function ct_os_new_project() {
     return
   fi
   local project_name="${1:-$(_ct_os_get_uniq_project_name)}" ; shift || :
-  oc new-project "${project_name}"
-  # let openshift cluster to sync to avoid some race condition errors
-  sleep 3
-  if test -n "${OPENSHIFT_CLUSTER_PULLSECRET_PATH:-}" -a -e "${OPENSHIFT_CLUSTER_PULLSECRET_PATH:-}"; then
-    oc create -f "$OPENSHIFT_CLUSTER_PULLSECRET_PATH"
-    # add registry pullsecret to the serviceaccount if provided
-    secret_name=$(grep '^\s*name:' "$OPENSHIFT_CLUSTER_PULLSECRET_PATH" | awk '{ print $2 }')
-    oc secrets link --for=pull default "$secret_name"
+  if [ "${SHARED_CLUSTER:-false}" == 'true' ]; then
+    oc create -f - <<EOF
+apiVersion: tenant.paas.redhat.com/v1alpha1
+kind: TenantNamespace
+metadata:
+  name: ${project_name}
+  namespace: core-services-ocp--config
+spec:
+  type: runtime
+  roles:
+    - namespace-admin
+    - tenant-egress-admin
+  network:
+    security-zone: internal
+EOF
+    sleep 3
+    oc apply -f - <<EOF
+apiVersion: tenant.paas.redhat.com/v1alpha1
+kind: TenantEgress
+metadata:
+    name: default
+    namespace: core-services-ocp--${project_name}
+spec:
+  egress:
+  - to:
+      dnsName: github.com
+    type: Allow
+  - to:
+      cidrSelector: 172.0.0.0/8
+    type: Allow
+  - to:
+      cidrSelector: 10.0.0.0/9
+    type: Allow
+  - to:
+      cidrSelector: 52.218.128.0/17
+    type: Allow
+  - to:
+      cidrSelector: 52.92.128.0/17
+    type: Allow
+  - to:
+      cidrSelector: 52.216.0.0/15
+    type: Allow
+EOF
+    export PROJECT_NAME="$project_name"
```

Comment:
PROJECT_NAME has to be exported because `project_name` has full name like `core-services-ocp--sclorg-<number>` and this full name cannot be deleted.